jparest/src/main/java/uk/gov/homeoffice/digital/sas/jparest/service/ResourceApiService.java

```diff
@@ -173,24 +87,24 @@ public void deleteResource(UUID tenantId, UUID id) {
     }
   }
 
-  public T updateResource(UUID tenantId, UUID id, T payload) {
+  public T updateResource(T entity) {
 
     var transactionDefinition = new DefaultTransactionDefinition();
     var transactionStatus = this.transactionManager.getTransaction(transactionDefinition);
-    T orig;
+    T originalEntity;
     try {
-      this.entityValidator.validateAndThrowIfErrorsExist(payload);
-      orig = repository.findById(id).orElseThrow(() -> new ResourceNotFoundException(id));
-      validateResourceTenantId(tenantId, orig, id);
-      BeanUtils.copyProperties(payload, orig, EntityUtils.ID_FIELD_NAME);
-      repository.saveAndFlush(orig);
+      this.entityValidator.validateAndThrowIfErrorsExist(entity);
+      originalEntity = repository.findByTenantIdAndId(entity.getTenantId(), entity.getId())
+              .orElseThrow(() -> new ResourceNotFoundException(entity.getId()));
+      BeanUtils.copyProperties(entity, originalEntity, EntityUtils.ID_FIELD_NAME);
+      repository.saveAndFlush(originalEntity);
       transactionManager.commit(transactionStatus);
     } catch (RuntimeException ex) {
       transactionManager.rollback(transactionStatus);
       throw ex;
     }
```

Comment:
This won't be needed if enclosing method is annotated with @Transactional

Reply:
Part of the next subtask eahw-2545